src/DotNetLightning.Core/Utils/NBitcoinExtensions.fs

```diff
@@ -35,6 +36,9 @@ module NBitcoinExtensions =
     type Key with
         static member BytesLength: int = 32
 
+        static member FromHash(preimage: array<byte>): Key =
+            Key (Hashes.SHA256 preimage)
```

Comment:
argh another `new` keyword missing, sorry for not spotting it earlier

Reply:
Ditto. However we really need to fix all the build warnings so that these things are easier to spot.

Fixed.

Reply:
well, I tried in one of my latest PRs, there was a warning that I didn't manage to get rid of... beyond my paygrade.